src/WebJobs.Script.WebHost/Management/LinuxSpecialization/RunFromPackageHandler.cs

```diff
@@ -0,0 +1,280 @@
+                {
+                    await _meshServiceClient.MountFuse(MeshServiceClient.ZipOperation, filePath, scriptPath);
+                }
+                else
+                {
+                    if (useLocalSitePackages)
+                    {
+                        _bashCommandHandler.UnzipPackage(filePath, localSitePackagesPath);
+                        await CreateBindMount(localSitePackagesPath, scriptPath);
+                    }
+                    else
+                    {
+                        _bashCommandHandler.UnzipPackage(filePath, scriptPath);
+                    }
+                }
+            }
+        }
+
+        private CodePackageType GetPackageType(string filePath, RunFromPackageContext pkgContext)
+        {
+            // cloud build always builds squashfs
+            if (pkgContext.IsScmRunFromPackage())
+            {
+                return CodePackageType.Squashfs;
+            }
+
+            var uri = new Uri(pkgContext.Url);
+            // check file name since it'll be faster than running `file`
+            if (FileIsAny(".squashfs", ".sfs", ".sqsh", ".img", ".fs"))
+            {
+                return CodePackageType.Squashfs;
+            }
+            else if (FileIsAny(".zip"))
+            {
+                return CodePackageType.Zip;
+            }
+
+            // Check file magic-number using `file` command.
+            (var output, _, _) = _bashCommandHandler.RunBashCommand($"{BashCommandHandler.FileCommand} -b {filePath}", MetricEventNames.LinuxContainerSpecializationFileCommand);
+            if (output.StartsWith("Squashfs", StringComparison.OrdinalIgnoreCase))
+            {
+                return CodePackageType.Squashfs;
+            }
+            else if (output.StartsWith("Zip", StringComparison.OrdinalIgnoreCase))
+            {
+                return CodePackageType.Zip;
+            }
+            else
+            {
+                throw new InvalidOperationException($"Can't find CodePackageType to match {filePath}");
+            }
+
+            bool FileIsAny(params string[] options)
+                => options.Any(o => uri.AbsolutePath.EndsWith(o, StringComparison.OrdinalIgnoreCase));
+        }
+
+        public async Task<string> Download(RunFromPackageContext pkgContext)
+        {
+            var zipUri = new Uri(pkgContext.Url);
+            if (!Utility.TryCleanUrl(zipUri.AbsoluteUri, out string cleanedUrl))
+            {
+                throw new Exception("Invalid url for the package");
+            }
+
+            var tmpPath = Path.GetTempPath();
+            var fileName = Path.GetFileName(zipUri.AbsolutePath);
+            var filePath = Path.Combine(tmpPath, fileName);
+            if (pkgContext.PackageContentLength != null && pkgContext.PackageContentLength > AriaDownloadThreshold)
+            {
+                _logger.LogInformation($"Downloading zip contents from '{cleanedUrl}' using aria2c'");
```

Comment:
As an improvement in future PRs - extract to a base class  and provide concrete implementation for each type of download we support - right now its Aria and HttpClient - this will make it easy to test and extend to other types download if needed.

Reply:
Sure.